### PR TITLE
Prevent re-toasting

### PR DESCRIPTION
**Pull request by tiltom**
Resolves https://sovryn.monday.com/boards/2218344956/pulses/2287699393

### CODE REVIEW

File: src/app/pages/PerpetualPage/slice.ts [hunk 1]
```diff
@@ -51,6 +52,9 @@ const perpetualPageSlice = createSlice({
     reset(state) {
       state = initialState;
     },
+    setToastedTransactions(state, { payload }: PayloadAction<string>) {
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/slice.ts:55]: set* implies changing the whole variable value, use push instead.
```suggestion
    pushToastedTransaction(state, { payload }: PayloadAction<string>) {
```

**Review by ShynRou — CHANGES_REQUESTED** [review]

**Review by soulBit — APPROVED** [review]

File: src/app/pages/PerpetualPage/components/ToastsWatcher/index.tsx [hunk 2]
```diff
@@ -66,12 +67,10 @@ export const ToastsWatcher: React.FC = () => {
           );
         }
       }
-      setToastedTransactions(transactions => ({
-        ...transactions,
-        [transaction.transactionHash]: true,
-      }));
+
+      dispatch(actions.pushToastedTransactions(transaction.transactionHash));
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/components/ToastsWatcher/index.tsx:71]: Sorry I don't want to be pedantic, :face_exhaling: **but**
```suggestion
      dispatch(actions.pushToastedTransaction(transaction.transactionHash));
```

File: src/app/pages/PerpetualPage/slice.ts [hunk 3]
```diff
@@ -51,6 +52,9 @@ const perpetualPageSlice = createSlice({
     reset(state) {
       state = initialState;
     },
+    pushToastedTransactions(state, { payload }: PayloadAction<string>) {
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/slice.ts:55]:
```suggestion
    pushToastedTransaction(state, { payload }: PayloadAction<string>) {
```

**Review by ShynRou — CHANGES_REQUESTED** [review]

File: src/app/pages/PerpetualPage/components/ToastsWatcher/index.tsx (hunk 2 above)
Comment (tiltom) [reply src/app/pages/PerpetualPage/components/ToastsWatcher/index.tsx:71]: Sorry, I was probably blind :D

**Review by ShynRou — APPROVED** [review]

**Pull request merged**